src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -344,6 +344,8 @@ public bool IsQueryKeyword(SyntaxToken token)
                         default:
                             return false;
                     }
+                case SyntaxKind.WhereKeyword:
+                    return token.Parent.Kind() == SyntaxKind.WhereClause; // false for e.g. type parameter constraint
```

Comment:
Iteration 11 is adding some logic, but I don't see any impacted tests. Are you planning to add such tests? Thanks

Reply:
AFIK there aren't any test for `CSharpSyntaxFactsService`. This service is tested indirectly by it's usage in other services (e.g. QuickInfo). @CyrusNajmabadi knows more about it.

Reply:
Yes.  We do not test this service directly.  We're ok with it being indirectly covered through tests of other features.  We want to ensure that the experience users have of features is what is tested.  We do not believe in testing the underlying implementation-detail pieces as that just creates brittle tests for us where we have to go update tests when impl-details change, even if the user experience stays the same.